File: src/uk/co/ben_gibson/git/link/Git/RemoteHost.java
package uk.co.ben_gibson.git.link.Git;

import com.intellij.openapi.util.IconLoader;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

public enum RemoteHost
{
    GIT_HUB("GitHub", "/Icons/GitHub/GitHub.png"),
    BITBUCKET_SERVER("Bitbucket Server", "/Icons/Bitbucket/Bitbucket.png"),
    BITBUCKET_CLOUD("Bitbucket Cloud", "/Icons/Bitbucket/Bitbucket.png"),
    GITLAB("GitLab", "/Icons/GitLab/GitLab.png"),
    GITBLIT("GitBlit", ""),
    GITEA("Gitea", "/Icons/Gitea/Gitea.png"),
    GOGS("Gogs", "/Icons/Gogs/Gogs.png"),
    CUSTOM("Custom", "/Icons/Custom/Custom.png");

    private final String displayName;
    private final String icon;

    RemoteHost(@NotNull String displayName, @NotNull String icon)
    {
        this.displayName = displayName;
        this.icon        = icon;
    }

    @NotNull
    public String toString()
    {
        return this.displayName;
    }

    public boolean isCustom()
    {
        return (this == CUSTOM);
    }

    public boolean isGitBlit()
    {
        return (this == GITBLIT);
    }

    public boolean isGitHub()
    {
        return (this == GIT_HUB);
    }

    public boolean isGitLab()
    {
        return (this == GITLAB);
    }

    public boolean isBitbucketCloud()
    {
        return (this == BITBUCKET_CLOUD);
    }

    public boolean isBitbucketServer()
    {
        return (this == BITBUCKET_SERVER);
    }

    public boolean isGitea()
    {
        return (this == GITEA);
    }

    public boolean isGogs()
    {
        return (this == GOGS);
    }

    @NotNull
    public Icon icon()
    {
        return IconLoader.getIcon(this.icon);
    }
}

File: src/main/kotlin/uk/co/ben_gibson/git/link/git/RemoteHost.kt
package uk.co.ben_gibson.git.link.git

import com.intellij.icons.AllIcons
import git4idea.repo.GitRemote
import java.net.URL
import javax.swing.Icon

enum class RemoteHost(val displayName: String, val icon: Icon, val defaultUrl: URL) {
    GIT_HUB("GitHub", AllIcons.Vcs.Vendors.Github, URL("https://github.com")),
    BITBUCKET_SERVER("Bitbucket Server", AllIcons.Vcs.Vendors.Github, URL("https://github.com")),
    BITBUCKET_CLOUD("Bitbucket Cloud", AllIcons.Vcs.Vendors.Github, URL("https://github.com")),
    GITLAB("GitLab", AllIcons.Vcs.Vendors.Github, URL("https://gitlab.com")),
    GITBLIT("GitBlit", AllIcons.Vcs.Vendors.Github, URL("https://github.com")),
    GITEA("Gitea", AllIcons.Vcs.Vendors.Github, URL("https://github.com")),
    GOGS("Gogs", AllIcons.Vcs.Vendors.Github, URL("https://github.com")),
    AZURE("Azure", AllIcons.Vcs.Vendors.Github, URL("https://github.com")),
    CUSTOM("Custom", AllIcons.Vcs.Vendors.Github, URL("https://github.com"));

    companion object {
        fun findHostByRemote(remote: GitRemote) = remote
            .httpUrl()
            ?.host
            ?.let { host -> values().first { it.defaultUrl.host == host } }
    }
}